instrumentation/ktor/ktor-2.0/testing/src/main/kotlin/io/opentelemetry/instrumentation/ktor/v2_0/client/AbstractKtorHttpClientTest.kt

```diff
@@ -71,4 +77,24 @@ abstract class AbstractKtorHttpClientTest : AbstractHttpClientTest<HttpRequestBu
       }
     }
   }
+
+  @Test
+  fun checkSpanEndsAfterBodyReceived() {
```

Comment:
Would it make sense to move this to `AbstractHttpClientTest` to ensure that other http client instrumentations also behave the same way?

Reply:
@marychatte can you confirm that the change conforms with this part of the spec also? thanks

> If there is any possibility for application code to not fully read the HTTP response (and for the HTTP client library to then have to clean up the HTTP response asynchronously), the HTTP client span SHOULD NOT be ended in this cleanup phase, and instead SHOULD end at some point after the HTTP response headers are fully read (or fail to be read). This avoids the span being ended asynchronously later on at a time which is no longer directly associated with the application code which made the HTTP request.

Reply:
@trask, there will be no **later** from `This avoids the span being ended asynchronously later`. Span will be ended right after the body is received. It's not the clean-up phase in the usual sense, we are using `invokeOnCompletion` at the moment of the ending of the body.

Reply:
thanks, what happens if a user reads the response headers and then decides not to read (ignore) the body?